lib/rom/http/dataset.rb

```diff
@@ -0,0 +1,103 @@
+module ROM
+  module HTTP
+    class Dataset
+      include Enumerable
+
+      attr_reader :config, :options
+
+      def initialize(config, options = {})
+        @config = config
+        @options = {
+          request_method: :get,
+          path: '',
+          params: {}
+        }.merge(options)
+      end
+
+      def url
+        config[:url]
```

Comment:
`url` probably can't be `nil` so maybe `config.fetch(:url)` and raise `ROM::HTTP` name spaced exception if missing?